### PR TITLE
Feature flag/fetch tests validators

### PR DESCRIPTION
**Pull request by akash2819**
added unit and integration tests for fetch flags

### CODE REVIEW

File: test/unit/models/flag.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  });
+
+  describe("fetch all flag", function () {
+    it("Should return all flags", async function () {
+      const fetched = await flagFirestore.doc(flagId).get();
+      const fetchedData = fetched.data();
+      Object.keys(flagData).forEach((key) => {
+        expect(flagData[key]).to.deep.equal(fetchedData[key]);
+      });
+    });
```
Comment (RahulGoyal-tech) [inline test/unit/models/flag.test.js:42]: You can add a test case when no flag was found or 0 flags were returned.

**Review by RahulGoyal-tech — CHANGES_REQUESTED** [review]

**Comment by pallabez** [comment]
Nice work with tests. :+1:

File: test/unit/models/flag.test.js (hunk 1 above)
Comment (akash2819) [reply test/unit/models/flag.test.js:47]: I have made changes please review them

File: test/integration/flag.test.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        .get("/flag")
+        .end((res, err) => {
+          if (err) {
+            return done();
+          }
+
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Flags returned successfully!");
+          expect(res.body.featureFlags).to.be.a("array");
```
Comment (pallabez) [inline test/integration/flag.test.js:46]: We should verify the content of the array also right?

File: test/unit/models/flag.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      const fetchedData = fetched.data();
+      const fetchedId = fetched.id;
+      Object.keys(flagData).forEach((key) => {
+        expect(flagData[key]).to.deep.equal(fetchedData[key]);
+      });
+      expect(fetchedId).to.equal(flagId);
+    });
+  });
+
+  describe("fetch all flag", function () {
```
Comment (pallabez) [inline test/unit/models/flag.test.js:35]: It should be the function name and not the description

File: test/unit/models/flag.test.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    it("Should return all flags", async function () {
+      const fetched = await flagFirestore.doc(flagId).get();
+      const fetchedData = fetched.data();
+      Object.keys(flagData).forEach((key) => {
+        expect(flagData[key]).to.deep.equal(fetchedData[key]);
+      });
+      expect(fetchedData).to.be.a("Object");
```
Comment (pallabez) [inline test/unit/models/flag.test.js:42]: We should be using the function to get the data.

File: test/unit/models/flag.test.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    });
+    it("Should return  no flag found", async function () {
+      const fetched = await flagFirestore.doc(flagId).get();
+      should.not.equal(fetched.id, 0);
```
Comment (pallabez) [inline test/unit/models/flag.test.js:46]: Same here. Use function `fetchFlags` to get flag data.

**Comment by prakashchoudhary07** [comment]
Closing this as the parent ticket is closed https://github.com/Real-Dev-Squad/website-backend/pull/570

**Pull request closed without merging**